provider/inspec.rb

```diff
@@ -62,10 +62,11 @@ def generate_resource(data)
 
     def generate_properties(data)
       object = data[:object]
-      nested_object_arrays = object.properties.select\
+      props = object.is_a?(::Api::Resource) ? object.all_user_properties : object.properties
```

Comment:
What would object be otherwise?

Reply:
This method is called recursively, as the nested properties of resources can have nested properties themselves. Only resources have all_user_properties though